ssc/cmod_utilityrate5.cpp

```diff
@@ -1775,24 +1828,194 @@ class cm_utilityrate5 : public compute_module
+        2=Single meter with no monthly rollover credits (Net Billing)
+        3=Single meter with monthly rollover credits in $ (Net Billing $)
+        4=Two meters with all generation sold and all load purchaseded
+        */
+        //int metering_option = as_integer("ur_metering_option");
+        int excess_dollars_credit_month = (int)as_number("ur_nm_credit_month");
+
+        bool tou_demand_single_peak = (as_integer("TOU_demand_single_peak") == 1);
 
 
 
+
+        // calculate the monthly net energy and monthly hours
+        
+        size_t surplus_tier = 0, deficit_tier = 0;
+        
+
+
 // main loop
 		c = 0;
 		// process one month at a time
 		for (m = 0; m < (int)rate.m_month.size(); m++)
 		{
 			ur_month& curr_month = rate.m_month[m];
+            monthly_deficit_energy = 0;
+            monthly_surplus_energy = 0;
+            if (ur_ec_hourly_acc_period == 1) {
+                surplus_tier = 0;
+                deficit_tier = 0;
+            }
 			if (curr_month.hours_per_month <= 0) break;
 			for (d = 0; d<util::nday[m]; d++)
 			{
-				for (h = 0; h<24; h++)
+                daily_deficit_energy = 0;
+                daily_surplus_energy = 0;
+                if (ur_ec_hourly_acc_period == 2) {
+                    surplus_tier = 0;
+                    deficit_tier = 0;
+                }
+                for (h = 0; h<24; h++)
 				{
 					// energy charge
 					for (s = 0; s < (int)steps_per_hour && c < (int)m_num_rec_yearly; s++)
 					{
-						if (d == util::nday[m] - 1 && h == 23 && s == (int)(steps_per_hour-1) )
+                        int row = rate.get_tou_row(c, m);
+                        step_surplus_energy = 0.0;
+                        step_deficit_energy = 0.0;
+
+                        if (e_in[c] >= 0.0)
+                        { // calculate income or credit
+                            e_upper = curr_month.ec_tou_ub.at(row, surplus_tier); // Have to check this each step to swap between surplus and deficit
+                            monthly_surplus_energy += e_in[c];
+                            daily_surplus_energy += e_in[c];
+
+                            // base period charge on units specified
+                            ssc_number_t energy_surplus = e_in[c];
+                            ssc_number_t cumulative_energy = e_in[c];
+                            if (ur_ec_hourly_acc_period == 1)
+                                cumulative_energy = monthly_surplus_energy;
+                            else if (ur_ec_hourly_acc_period == 2)
+                                cumulative_energy = daily_surplus_energy;
+
+
+                            // cumulative energy used to determine tier for credit of entire surplus amount
+                            ssc_number_t credit_amt = 0;
+
+                            ssc_number_t tier_credit = 0.0, sr = 0.0, tier_energy = 0.0;
+                            // time step sell rates
+                            if (as_boolean("ur_en_ts_sell_rate")) {
```

Comment:
This should always be false. ur_calc only runs for metering options 0 and 1, and the GUI disables time series rates for those options.